core-java-modules/core-java-numbers-6/src/test/java/com/baeldung/javadoublevsbigdecimal/BigDecimalUnitTest.java

```diff
@@ -0,0 +1,50 @@
+package com.baeldung.javadoublevsbigdecimal;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class BigDecimalUnitTest {
+    private BigDecimal bd1;
+    private BigDecimal bd2;
+    private BigDecimal expected;
+    private BigDecimal actual;
+
+    @BeforeEach
+    public void setup() {
+        bd1 = new BigDecimal("124567890.0987654321");
+        bd2 = new BigDecimal("987654321.123456789");
```

Comment:
Since these values don't change there's no need to create them for every test.

Reply:
I mean there's no need to use @BeforeEach

Reply:
Oh right. Got that. Fixed it in the article draft too